packages/desktop-client/src/components/reports/ReportLegend.tsx

```diff
@@ -31,37 +31,38 @@ function ReportLegend({ legend, groupBy }: ReportLegendProps) {
         {groupBy}
       </Text>
       <View>
-        {legend.map(item => {
-          return (
-            <View
-              key={item.name}
-              style={{
-                padding: 10,
-                flexDirection: 'row',
-                alignItems: 'center',
-              }}
-            >
+        {legend &&
+          legend.map(item => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏is `legend` optional? you'll want to update the prop type too then to reflect this better.

Reply:
I've added an error check instead. LMK if this works.

Reply:
I think we have a misunderstanding. Your previous logic was totally fine, but the component prop type needs to reflect the new reality. Let me explain a bit more in depth.

```
// This component always requires a `test` prop to be set
function MyComponent(props: { test: boolean }) {}


// This component makes `test` prop optional
function MyComponent(props: { test?: boolean }) {}

// This is what you currently have:
type ReportLegendProps = {
  // notice that the `legend` prop is ALWAYS required
  legend: Array<{ name: string; color: string }>;
  groupBy: string;
}

// Instead it should now be
type ReportLegendProps = {
  legend?: Array<{ name: string; color: string }>;
  groupBy: string;
```